src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3691,6 +3718,7 @@ private static bool HasNonConstantInitializer(ArrayBuilder<ArrayBuilder<FieldOrP
 
                             var constructor = SourceConstructorSymbol.CreateConstructorSymbol(this, constructorSyntax, diagnostics);
                             builder.NonTypeNonIndexerMembers.Add(constructor);
+                            builder.UpdateIsNullableEnabled(useStatic: constructor.IsStatic, compilation, constructorSyntax);
```

Comment:
>builder.UpdateIsNullableEnabled(useStatic: constructor.IsStatic, compilation, constructorSyntax); [](start = 28, length = 97)

It feels like this shouldn't be called for instance constructors that have ```this``` initializers. Basically those that return false for ```IncludeFieldInitializersInBody``` should probably be skipped. Perhaps the work for constructors to contribute to the shared state should be postponed till later.